driver/reducecalculation_driver.hpp

```diff
@@ -136,18 +173,20 @@ class ReduceCalculationDriver : public Driver
     std::unique_ptr<GPUMem> workspace_dev;
 
     std::vector<Tgpu> in;
-    std::vector<Tgpu> out;
+    std::vector<Tgpu_out> out;
     std::vector<Tref> outhost;
 
     size_t ws_sizeInBytes;
 
     int dim;
     miopenReduceCalculationNanPropagation_t nanPropagation;
     miopenReduceCalculationOp_t reduceCalculationOp;
+
+    bool isLogicalCalculation = false;
 };
 
-template <typename Tgpu, typename Tref>
-int ReduceCalculationDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+template <typename Tgpu, typename Tref, typename Tgpu_out>
+int ReduceCalculationDriver<Tgpu, Tref, Tgpu_out>::ParseCmdLineArgs(int argc, char* argv[])
```

Comment:
same

Reply:
Please check my new implementation, I removed Tgpu_out as it's unnecessary